src/Http/Authentication.Abstractions/src/AuthenticationOptions.cs

```diff
@@ -113,4 +113,9 @@ public AddSchemeState([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes
     /// A user is considered authenticated if <see cref="ClaimsIdentity.IsAuthenticated"/> returns <see langword="true" /> for the <see cref="ClaimsPrincipal"/> associated with the HTTP request.
     /// </summary>
     public bool RequireAuthenticatedSignIn { get; set; } = true;
+
+    /// <summary>
+    /// If true, DefaultScheme will not automatically use a single registered scheme.
+    /// </summary>
+    public bool DisableAutoDefaultScheme { get; set; }
```

Comment:
Would we consider making this an AppContext switch for one release to see if anybody ever needs it?

Reply:
Is that in practice better than just having a obscure public flag that no one uses?

Reply:
The flag lives forever, even if nobody needs it. A switch we can remove if nobody uses it.

@DamianEdwards ?

Reply:
It's true we can avoid the public API addition if it turns out nobody uses the `AppContext` switch for a release. I'm supportive of changing this to a switch instead and we can always add the option if we decide it's worthwhile.

Reply:
Sounds fine as long as we have a clear idea of how to make this determination in 8.0, or are we just going to remove it and see if anyone complains in 8?

Reply:
We'll see if anyone complains in 7 and we have to tell them how to turn on the switch. I'd even leave the switch out of the announcement to encourage them to contact us.

Reply:
Don’t leave it out if the announcement

Reply:
Just a thought, why not neither have the property nor the `AppContext` switch to save one if check! and whoever needed the old behavior could add a dummy authentication scheme which never authenticate. :confused:

Reply:
So one argument against the appcontext switch is that it really creates a headache testing, as our xunit tests run in parallel and it seems like setting this switch affects all the tests, so even if I set the switch via a try/finally, I'm seeing tests fail due to this.

They run fine if I run the tests sequentially, but when running the full set via dotnet test, I'll see failures.

How do we feel about just keeping the public option?

Reply:
We can do both an internal option for testing and an appcontext switch for everyone else. For example:

https://github.com/dotnet/aspnetcore/blob/d96a100bddc72606f7417b665428411388b8ac54/src/Servers/Kestrel/Core/src/KestrelServerOptions.cs#L160-L176

Reply:
Cool that's not too bad then as the code will basically stay the same in this PR, just an additional app context check in the get for the options property

Reply:
Note I still think this is a net negative, as now we have internals visible to 3 more assemblies, when we could have just had this be a public bool

Reply:
I'll leave it you ya'll to decide, but I tend to agree with @HaoK in this case that it seems like a lot of trouble to avoid a single public `bool` property.